theseus/__init__.py

```diff
@@ -45,3 +45,8 @@
 from .theseus_layer import TheseusLayer
 
 import theseus.embodied as eb
+
+import torch
+
+if torch.cuda.is_available():
```

Comment:
Should we have an else here that raises an error/warning, if the user tries to call/use `LUCudaSparseSolver` without having cuda.

Reply:
If cuda is not available, then `LUCudaSparseSolver` would not even be visible, so trying to do `import th.LUCudaSparseSolver` would throw an error. Maybe this will be confusing for users?

Is it possible to make all of these imports permanent in the relevant `__init__.py` files, but address the `cuda.is_available()` issue inside the corresponding `.py` files? Basically, allow users to import `LUCudaSparseSolver` and only throw an error if this class is attempted to be instantiated w/o cuda being available.

Reply:
Throwing an error seems okay but your other suggestion is also good. Not sure if there is a standard convention here that we can follow.

Reply:
Right, but the error will be something like:

```python
import th
solver = th.LUCudaSparseSolver(...)
...
...
AttributeError: module 'theseus' has no attribute 'LUCudaSparseSolver'
```

which seems misleading, no?

Reply:
What we want is for this to be a `RuntimeError` that says `"class `LUCudaSparseSolver` requires cuda to be available"`. This could be achieved, for example, by adding the check inside the class' constructor, or, if that's not possible due to some C++ import issues, making a stub class with no requirements whose job is only to throw an error, and that gets imported in place of the original one when no cuda is present.

Reply:
Yes, we would want an error message that indicates this functionality is only available with cuda. Whichever version get us there easily sounds good.

Reply:
ok but at the level of the python class right? so no problems creating a mock python class which just throws error then I think? This may confuse python typing unless the mock class is mimicking all the methods with the same types.

Reply:
Oh, but that's a good point about typing and the mock class. Is there a reason why the error cannot be thrown inside the original class' constructor? Can you elaborate a bit?

Reply:
so, the problem is that if cuda is not available I cannot import any C++ extension depending on cuda either (or an error occurs). Therefore the "import" has to be done inside the functions themselves. Do you see any issue with that?